openapi/commercetools.productcatalog/client.bal

```diff
@@ -0,0 +1,638 @@
+public type ClientConfig record {|
+    # Configurations related to client authentication
+    http:OAuth2ClientCredentialsGrantConfig|http:BearerTokenConfig auth;
+    # The HTTP version understood by the client
+    string httpVersion = "1.1";
+    # Configurations related to HTTP/1.x protocol
+    http:ClientHttp1Settings http1Settings = {};
+    # Configurations related to HTTP/2 protocol
+    http:ClientHttp2Settings http2Settings = {};
+    # The maximum time to wait (in seconds) for a response before closing the connection
+    decimal timeout = 60;
+    # The choice of setting `forwarded`/`x-forwarded` header
+    string forwarded = "disable";
+    # Configurations associated with Redirection
+    http:FollowRedirects? followRedirects = ();
+    # Configurations associated with request pooling
+    http:PoolConfiguration? poolConfig = ();
+    # HTTP caching related configurations
+    http:CacheConfig cache = {};
+    # Specifies the way of handling compression (`accept-encoding`) header
+    http:Compression compression = http:COMPRESSION_AUTO;
+    # Configurations associated with the behaviour of the Circuit Breaker
+    http:CircuitBreakerConfig? circuitBreaker = ();
+    # Configurations associated with retrying
+    http:RetryConfig? retryConfig = ();
+    # Configurations associated with cookies
+    http:CookieConfig? cookieConfig = ();
+    # Configurations associated with inbound response size limits
+    http:ResponseLimitConfigs responseLimits = {};
+    # SSL/TLS-related options
+    http:ClientSecureSocket? secureSocket = ();
+|};
+
+# This is a generated connector for [Commercetools API v1](https://docs.commercetools.com/api/) OpenAPI specification. 
+# The commercetools Platform HTTP API(Product Catalog) offers an interface for programmatic access to the data stored on the platform and associated functionality for product catalogs.
+@display {label: "Commercetools Product Catalog", iconPath: "icon.png"}
+public isolated client class Client {
+    final http:Client clientEp;
+    # Gets invoked to initialize the `connector`.
+    # The connector initialization requires setting the API credentials. Create and configure an OAuth2 client credentials by following [this guide](https://docs.commercetools.com/api/authorization).
+    #
+    # + clientConfig - The configurations to be used when initializing the `connector` 
+    # + serviceUrl - URL of the target service 
+    # + return - An error if connector initialization failed 
+    public isolated function init(ClientConfig clientConfig, string serviceUrl) returns error? {
+        http:Client httpEp = check new (serviceUrl, clientConfig);
+        self.clientEp = httpEp;
+        return;
+    }
+    #
+    # + return - 200 
+    remote isolated function getCategoryByID(string projectKey, string iD, Expansion[]? expand = ()) returns Category|error {
+        string resourcePath = string `/${projectKey}/categories/${iD}`;
```

Comment:
```suggestion
    remote isolated function getCategoryByID(string projectKey, string id, Expansion[]? expand = ()) returns Category|error {
        string resourcePath = string `/${projectKey}/categories/${id}`;
```

This can be achieved by updating the OpenAPI as below 
```yaml
- name: id
        in: path
        required: true
        schema:
          type: string
```

Reply:
We have updated their operation names only (correct me if I am wrong @SanduDS). If we update it will become bit of custom thing. In current way also functionality will be same and working right?

Reply:
Yes @abeykoon , HTTP API[1] shows ID as `id` but the openAPI spec shows as `ID`. our tool makes the 1st letter simple when it generates. so it becomes 'iD'. therefore, Let's go with the official API. 
[1] https://docs.commercetools.com/api/